code/modules/hydroponics/hydroponics_chemreact.dm

```diff
@@ -0,0 +1,226 @@
+/**
+  *This is NOW the gradual affects that each chemical applies on every process() proc. Nutrients now use a more robust reagent holder in order to apply less insane
+  * stat changes as opposed to 271 lines of individual statline effects. Shoutout to the original comments on chems, I just cleaned a few up.
+  */
+/obj/machinery/hydroponics/proc/applyChemicals(datum/reagents/S, mob/user)
+	if(myseed)
+		myseed.on_chem_reaction(S) //In case seeds have some special interactions with special chems, currently only used by vines
+
+	// Plants need at least a half unit in order to gain mutagenic affects, and prevent powergaming with dilluting chems.
+	if(S.has_reagent(/datum/reagent/toxin/mutagen, 0.5) || S.has_reagent(/datum/reagent/uranium/radium, 1) || S.has_reagent(/datum/reagent/uranium, 1))
+		switch(rand(100))
+			if(91 to 100)
+				adjustHealth(-10)
+				visible_message("<span class='warning'>[myseed.plantname] starts to wilt and burn!</span>")
+			if(41 to 90)
+				if(myseed && !self_sustaining) //Stability
+					myseed.adjust_instability(5)
+			if(21 to 40)
+				visible_message("<span class='notice'>[myseed.plantname] appears unusually reactive...</span>")
+			if(11 to 20)
+				mutateweed()
+			if(1 to 10)
+				mutatepest(user)
+
+	//Mutagenic chem side-effects.
+	if(S.has_reagent(/datum/reagent/uranium, 1))
+		adjustHealth(-round(S.get_reagent_amount(/datum/reagent/uranium) * 1))
+		adjustToxic(round(S.get_reagent_amount(/datum/reagent/uranium) * 2))
+	if(S.has_reagent(/datum/reagent/uranium/radium, 1))
```

Comment:
Maybe do these checks first, storing them in a `has_mutation_chem` kind of variable, to avoid repeating them?